pmultiqc/modules/common/file_utils.py

```
@@ -53,14 +53,28 @@ def extract_zip(file_path: str, extract_to: str) -> None:
-            # Normalize the path and check for path traversal attempts
-            member_path = os.path.normpath(os.path.join(extract_to, member))
-            if not member_path.startswith(os.path.abspath(extract_to)):
-                raise ValueError(f"Attempted path traversal in zip file: {member}")
-            if ".." in member or member.startswith("/"):
+            # Check for explicit path traversal attempts (../ or ..\ patterns)
+            # Allow filenames containing ".." as a substring (e.g., "file..name.txt")
+            if "../" in member or "..\\" in member or member.startswith("../") or member.startswith("..\\"):
                 raise ValueError(f"Invalid path in zip file: {member}")
+            # Check for absolute paths
+            if member.startswith("/") or (os.name == "nt" and len(member) > 1 and member[1] == ":"):
+                raise ValueError(f"Invalid absolute path in zip file: {member}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)